[BREAKING CHANGE] Use `robot_description` topic instead of `~/robot_description` and update docs regarding this (#1410)
ros-controls/ros2_control: controller_manager/doc/userdoc.rst

  Name of a plugin exported using ``pluginlib`` for a controller.
  This is a class from which controller's instance with name "``controller_name``" is created.

Subscribers
-----------

robot_description (std_msgs/msg/String)

>   String with the URDF string as robot description.

Reviewer: ```suggestion
  The URDF string as robot description.
```

  This is usually published by the ``robot_state_publisher`` node.

> 

Author: I'm not sure whether this is the way we want to suggest? Intuitively, I would use namespaces for handling this:

```python
control_node = Node(                                                                                                                                                                                                                                                               
    package="controller_manager",                                                                                                                                                                                                                                                  
    executable="ros2_control_node",                                                                                                                                                                                                                                                
    parameters=[robot_controllers],                                                                                                                                                                                                                                                
    output="both",                                                                                                                                                                                                                                                                 
    namespace="rrbot",                                                                                                                                                                                                                                                             
)                                                                                                                                                                                                                                                                                  
robot_state_pub_node = Node(                                                                                                                                                                                                                                                       
    package="robot_state_publisher",                                                                                                                                                                                                                                               
    executable="robot_state_publisher",                                                                                                                                                                                                                                            
    output="both",                                                                                                                                                                                                                                                                 
    parameters=[robot_description],                                                                                                                                                                                                                                                
    namespace="rrbot",                                                                                                                                                                                                                                                             
)
```

And when doing things with remaps, I would skip the initial `/` as this would only be true if the `controller_manager` is `/controller_manager` which is probably not the case in situations where we have multiple descriptions and controller managers:

```python
control_node = Node(                                                                                                                                                                                                                                                               
    package="controller_manager",                                                                                                                                                                                                                                                  
    executable="ros2_control_node",                                                                                                                                                                                                                                                
    parameters=[robot_controllers],                                                                                                                                                                                                                                                
    output="both",                                                                                                                                                                                                                                                                 
    remappings=[('robot_description', '/rrbot/robot_description')]                                                                                                                                                                                                                 
)                                                                                                                                                                                                                                                                                  
robot_state_pub_node = Node(                                                                                                                                                                                                                                                       
    package="robot_state_publisher",                                                                                                                                                                                                                                               
    executable="robot_state_publisher",                                                                                                                                                                                                                                            
    output="both",                                                                                                                                                                                                                                                                 
    parameters=[robot_description],                                                                                                                                                                                                                                                
    namespace="rrbot",                                                                                                                                                                                                                                                             
)
```

However, I could also add both versions to the docs.

> 

Reviewer: I agree with you on the initial `/`, I've updated my suggestion.  I thought of the namespace part, but in most cases, the robot_state_publisher node might not exist in the same launch file as that would give a lot of modularity. However, it is good to point this case as well as you suggested

> 

Author: I've updated things in [5c89eb3](https://github.com/ros-controls/ros2_control/pull/1410/commits/5c89eb3d04a24a8c6efb97b7cd033988a83022e4).

  Name of a plugin exported using ``pluginlib`` for a controller.
  This is a class from which controller's instance with name "``controller_name``" is created.

Subscribers
-----------

robot_description (std_msgs/msg/String)
  The URDF string as robot description.
  This is usually published by the ``robot_state_publisher`` node.

Handling Multiple Controller Managers
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> When dealing with multiple controller managers, you can manage different robot descriptions either
> using remappings or putting both, the ``robot_state_publisher`` and the ``controller_manager`` into
> the same namespace.
> 
> Using namespaces a snippet of a launchfile could look like the following:
> 
> .. code-block:: python
> 
>    control_node = Node(
>        package="controller_manager",
>        executable="ros2_control_node",
>        parameters=[robot_controllers],
>        output="both",
>        namespace="rrbot",
>    )
>    robot_state_pub_node = Node(
>        package="robot_state_publisher",
>        executable="robot_state_publisher",
>        output="both",
>        parameters=[robot_description],
>        namespace="rrbot",
>    )
> 
> 
> Using remappings:

Reviewer: ```suggestion
When dealing with multiple controller managers, you have two options for managing different robot descriptions:

Using namespaces a snippet of a launchfile could look like the following:

1. **Using Namespaces:** You can place both the ``robot_state_publisher`` and the ``controller_manager`` nodes into the same namespace. 

.. code-block:: python

   control_node = Node(
       package="controller_manager",
       executable="ros2_control_node",
       parameters=[robot_controllers],
       output="both",
       namespace="rrbot",
   )
   robot_state_pub_node = Node(
       package="robot_state_publisher",
       executable="robot_state_publisher",
       output="both",
       parameters=[robot_description],
       namespace="rrbot",
   )

2. **Using Remappings:** You can use remappings to handle different robot descriptions. This involves relaying topics using the ``remappings`` tag, allowing you to specify custom topics for each controller manager.
```

I restructured it a bit, if you don't like, it's fine :)

Author: That seems like a good idea, but then we should probably also drop line 94 "Using namespaces a snippet of a launchfile could look like the following:".

@saikishor if you could modifiy your suggestion, I could apply it directly :-)

Reviewer: Sure! on it

Reviewer: Done!

.. code-block:: python

   control_node = Node(

>        package="controller_manager",
> 

Reviewer: ```suggestion
       package="controller_manager",
       executable="ros2_control_node",
```

Here the executable is missing
